lib/puppet/file_serving/http_metadata.rb

```diff
@@ -0,0 +1,29 @@
+require 'puppet/file_serving/metadata'
+
+# Simplified metadata representation, suitable for the information
+# that is available from HTTP headers.
+class Puppet::FileServing::HttpMetadata < Puppet::FileServing::Metadata
+
+  def initialize(http_response, path = '/dev/null')
+    super(path)
+
+    # ignore options that do not apply to HTTP metadata
+    @owner = @group = @mode = nil
+
+    if checksum = http_response['content-md5']
+      # convert base64 digest to hex
+      checksum = checksum.unpack("m0").first.unpack("H*").first
+      @checksum_type = 'md5'
+      @checksum = "{md5}#{checksum}"
+    elsif mtime = DateTime.httpdate(http_response['last-modified']).to_time
+      @checksum_type = 'mtime'
+      @checksum = "{mtime}#{mtime}"
+    else
+      raise PuppetError, "HTTP response contained no usable checksum equivalent"
+    end
+
+    @ftype = 'file'
```

Comment:
It's that "unless we implemented x" that I'd like to avoid breaking or complicating in the future. I'd rather prevent anyone ever saying `ensure => directory`, rather than having it silently interpret that as `file` - until it doesn't.